### DISCUSSION TITLE
Blur out the background of dialog specifically a table or div in the background

### DISCUSSION THREAD

**Discussion by vinuthakaranth** [microsoft/fluentui#30409] [Q&A] [answered]
Hi Team,

Is there a way to specifically blur out the background of dialog something like below? Background can be a div or table.

![image](https://github.com/microsoft/fluentui/assets/33799331/51b01d9d-4c91-4baa-b30a-5b27a6989caa)

Also make a dialog appear just on part of page specifically on a div.

I tried using DialogSurface as in doc below but didn't apply to the backdrop. I might be missing something in implementation. Any help or direction to proceed with this is really appreciated. 
https://react.fluentui.dev/?path=/docs/components-dialog--default

I tried simple example by setting backdrop but didn't work as expected. 

[Codesandbox link](https://codesandbox.io/p/sandbox/happy-newton-fcvny2?file=%2Fsrc%2FApp.tsx%3A10%2C3&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clruxw1ad00063j6f43dolruc%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clruxw1ac00023j6fcmr7amo3%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clruxw1ac00033j6flqy1heks%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clruxw1ac00053j6fnq811sf7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clruxw1ac00023j6fcmr7amo3%2522%253A%257B%2522id%2522%253A%2522clruxw1ac00023j6fcmr7amo3%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrv14u7a00023j6knssq4a7o%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A10%252C%2522startColumn%2522%253A3%252C%2522endLineNumber%2522%253A10%252C%2522endColumn%2522%253A3%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FApp.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrv14u7a00023j6knssq4a7o%2522%257D%252C%2522clruxw1ac00053j6fnq811sf7%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clruxw1ac00043j6fj04wwq0v%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clruxw1ac00053j6fnq811sf7%2522%252C%2522activeTabId%2522%253A%2522clruxw1ac00043j6fj04wwq0v%2522%257D%252C%2522clruxw1ac00033j6flqy1heks%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clruxw1ac00033j6flqy1heks%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

If there is any example or doc I can refer too would be really helpful.

**Comment by khmakoto** [comment]
Hi @vinuthakaranth, you were on the right track! The only thing you missed is that passing JSX to a slot renders that as the `children` of the slot tag, so you were missing having the new backdrop div take all the `width` and `height` of its parent. You can see it working here.

https://codesandbox.io/p/sandbox/dialogsurface-with-custom-backdrop-3nhfks

**Reply by vinuthakaranth** [reply]
Thanks a lot @khmakoto for the quick response! I have a follow up question, is there a way to make dialog and backdrop to appear just on part of the page like on just a detail list/div in sandbox example? Is there any prop I can use directly? Thanks in advance for your help.

[Codesandbox link](https://codesandbox.io/p/sandbox/happy-newton-fcvny2?file=%2Fsrc%2Fexample.tsx%3A13%2C58&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clruxw1ad00063j6f43dolruc%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clruxw1ac00023j6fcmr7amo3%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clruxw1ac00033j6flqy1heks%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clruxw1ac00053j6fnq811sf7%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clruxw1ac00023j6fcmr7amo3%2522%253A%257B%2522id%2522%253A%2522clruxw1ac00023j6fcmr7amo3%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clrws0fv1003m3j6eu4vitxcs%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A13%252C%2522startColumn%2522%253A58%252C%2522endLineNumber%2522%253A13%252C%2522endColumn%2522%253A58%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252Fexample.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clrws0fv1003m3j6eu4vitxcs%2522%257D%252C%2522clruxw1ac00053j6fnq811sf7%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clruxw1ac00043j6fj04wwq0v%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522path%2522%253A%2522%252F%2522%257D%255D%252C%2522id%2522%253A%2522clruxw1ac00053j6fnq811sf7%2522%252C%2522activeTabId%2522%253A%2522clruxw1ac00043j6fj04wwq0v%2522%257D%252C%2522clruxw1ac00033j6flqy1heks%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clruxw1ac00033j6flqy1heks%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Reply by khmakoto** [reply, marked as answer]
@vinuthakaranth and I talked about this offline and it seemed like she wanted this same thing but on v8, after some back and forth we landed on the following example that covers everything she needed.

https://codepen.io/khmakoto/pen/WNmdWjw?editors=0010
